processor/src/test/resources/fixtures/org/mapstruct/ap/test/value/exception/CustomDefaultExceptionDefinedInMapperConfigImpl.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright MapStruct Authors.
+ *
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct.ap.test.value.exception;
+
+import javax.annotation.Generated;
+import org.mapstruct.ap.test.value.CustomIllegalArgumentException;
+import org.mapstruct.ap.test.value.ExternalOrderType;
+import org.mapstruct.ap.test.value.OrderType;
+
+@Generated(
+    value = "org.mapstruct.ap.MappingProcessor",
+    date = "2020-08-29T09:36:57+0200",
+    comments = "version: , compiler: javac, environment: Java 11.0.2 (AdoptOpenJDK)"
+)
+public class CustomDefaultExceptionDefinedInMapperConfigImpl implements CustomDefaultExceptionDefinedInMapperConfig {
+
+    @Override
+    public ExternalOrderType withAnyUnmapped(OrderType orderType) {
+        if ( orderType == null ) {
+            return null;
+        }
+
+        ExternalOrderType externalOrderType;
+
+        switch ( orderType ) {
+            default: externalOrderType = ExternalOrderType.DEFAULT;
+        }
+
+        return externalOrderType;
+    }
+
+    @Override
+    public ExternalOrderType withAnyRemaining(OrderType orderType) {
+        if ( orderType == null ) {
+            return null;
+        }
+
+        ExternalOrderType externalOrderType;
+
+        switch ( orderType ) {
+            case RETAIL: externalOrderType = ExternalOrderType.RETAIL;
+            break;
+            case B2B: externalOrderType = ExternalOrderType.B2B;
+            break;
+            default: externalOrderType = ExternalOrderType.DEFAULT;
+        }
+
+        return externalOrderType;
+    }
+
+    @Override
+    public ExternalOrderType onlyWithMappings(OrderType orderType) {
+        if ( orderType == null ) {
+            return null;
+        }
+
+        ExternalOrderType externalOrderType;
+
+        switch ( orderType ) {
+            case EXTRA: externalOrderType = ExternalOrderType.SPECIAL;
+            break;
+            case STANDARD: externalOrderType = ExternalOrderType.DEFAULT;
+            break;
+            case NORMAL: externalOrderType = ExternalOrderType.DEFAULT;
+            break;
+            case RETAIL: externalOrderType = ExternalOrderType.RETAIL;
+            break;
+            case B2B: externalOrderType = ExternalOrderType.B2B;
+            break;
+            default: throw new CustomIllegalArgumentException( "Unexpected enum constant: " + orderType );
```

Comment:
when you throw a checked exception.. shouldn't we check whether the method signature has defined such exception in the throws clause or do we leave this up to the compiler?

Reply:
I say we leave it to the compiler. In the end you as a user defines the exception that you want to throw, so it is entirely in your hands.